modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/repository/SnapshotPersister.java

```diff
@@ -302,8 +303,16 @@ public void persistComponent(DistributionSnapshot snapshot, ComponentInfo ci, Co
         for (ExtensionPointInfo epi : ci.getExtensionPoints()) {
             createExtensionPointDoc(snapshot, session, label, epi, componentDoc);
         }
+        Map<String, Integer> comps = new HashMap<String, Integer>();
         for (ExtensionInfo ei : ci.getExtensions()) {
-            createContributionDoc(snapshot, session, label, ei, componentDoc);
+            // handle multiple contributions to the same extension point
+            String id = ei.getId();
+            if (comps.containsKey(id)) {
+                comps.put(id, comps.get(id) + 1);
+            } else {
+                comps.put(id, Integer.valueOf(0));
+            }
```

Comment:
Could be simplified to:
```java
String id = ei.getId();
comps.computeIfPresent(id, (k,v) -> v + 1);
comps.computeIfAbsent(id, k -> Integer.valueOf(0));
```

Reply:
done, thanks, any suggestion about the similar change in RuntimeSnapshot?

Reply:
Not really as you need to change the id in one of the two cases.

Reply:
FYI @kevinleturc see my review for simpler code here